torchx/schedulers/slurm_scheduler.py

```diff
@@ -540,46 +562,79 @@ def _describe_sacct(self, app_id: str) -> Optional[DescribeAppResponse]:
-                roles_statuses[role] = RoleStatus(role, [])
-            roles[role].num_replicas += 1
-            roles_statuses[role].replicas.append(
-                ReplicaStatus(
-                    id=int(replica_id), role=role, state=app_state, hostname=""
+        roles: dict[str, Role] = {}
+        roles_statuses: dict[str, RoleStatus] = {}
+        state = AppState.UNKNOWN
+
+        for job in jobs:
+            # job name is of the form "{role_name}-{replica_id}"
+            role_name, _, replica_id = job["name"].rpartition("-")
+
+            entrypoint = job["command"]
+            image = job["current_working_directory"]
+            state = appstate_from_slurm_state(job["job_state"][0])
+
+            job_resources = job["job_resources"]
+
+            role = roles.setdefault(
+                role_name,
+                Role(
+                    name=role_name,
+                    image=image,
+                    entrypoint=entrypoint,
+                    num_replicas=0,
                 ),
             )
+            role_status = roles_statuses.setdefault(
+                role_name,
+                RoleStatus(role_name, replicas=[]),
+            )
+
+            # NOTE: torchx schedules on slurm with sbatch + heterogenous job
+            # where each replica is a "sub-job" so `allocated_nodes` will always be 1
+            # but we deal with jobs that have not been launched with torchx
+            # which can have multiple hosts per sub-job (count them as replicas)
+            node_infos = job_resources["allocated_nodes"]
+
+            for node_info in node_infos:
+                # NOTE: we expect resource specs for all the nodes to be the same
+                # NOTE: use allocated (not used/requested) memory since
+                #  users may only specify --cpu, in which case slurm
+                #  uses the (system) configured {mem-per-cpu} * {cpus}
+                #  to allocate memory.
+                # NOTE: getting gpus is tricky because it modeled as a trackable-resource
+                #  or not configured at all (use total-cpu-on-host as proxy for gpus)
+                cpu = int(node_info["cpus_used"])
+                memMB = int(node_info["memory_allocated"])
+
+                hostname = node_info["nodename"]
+
+                role.resource = Resource(cpu=cpu, memMB=memMB, gpu=-1)
+                role.num_replicas += 1
+                role_status.replicas.append(
+                    ReplicaStatus(
+                        id=int(replica_id),
+                        role=role_name,
+                        state=state,
+                        hostname=hostname,
+                    )
+                )
 
         return DescribeAppResponse(
             app_id=app_id,
             roles=list(roles.values()),
             roles_statuses=list(roles_statuses.values()),
-            state=app_state,
-            msg=msg,
```

Comment:
msg isn't needed?

Reply:
yea `msg` defaults to an empty string if not specified. We were just setting `msg=state` so no real functional value added + describe_sacct didn't set msg